### PR TITLE
Add documentations for the project

### PR DESCRIPTION
**Pull request by B1ueber2y**
Documentations were added with Sphinx

Format all the comments for the interface. Documentations are available here: http://b1ueber2y.me/projects/LIMAP/docs/

### CODE REVIEW

File: docs/tutorials/localization.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
+Localization with points & lines
+=================================
+
+Currently, runner scripts are provided to run visual localization integrating line along with point features on the following Datasets: 
+
+* `7Scenes Dataset <https://www.microsoft.com/en-us/research/project/rgb-d-dataset-7-scenes/>`_
+* `Cambridge Landmarks Dataset <https://www.repository.cam.ac.uk/handle/1810/251342>`_
```
Comment (rpautrat) [inline docs/tutorials/localization.rst:7]: The link to the Cambridge dataset redirects me to a blank page. Maybe we can instead put the link to the original paper: https://arxiv.org/pdf/1505.07427v4.pdf

**Review by MarkYu98 — APPROVED** [review]
Looks super duper good

File: docs/tutorials/visualization.rst [hunk 2]
```diff
@@ -0,0 +1,12 @@
+Visualization
+=================================
+
+We provide the visualization interface at `here <https://github.com/cvg/limap/blob/main/visualize_3d_lines.py>`_ to visualize the reconstructed line maps. Optionally, the camera frustums can also be visualized with an additional :class:`limap.base.ImageCollection` instance as the input. An example visualization command could be:
```
Comment (rpautrat) [inline docs/tutorials/visualization.rst:4]: Remove the "at" in "at here".

**Review by rpautrat — APPROVED** [review]
I added two minor comments that you can fix, but otherwise the doc looks good to me!

**Pull request merged**